Automattic/jetpack
# ActivityPub: add compatibility functions (targeting Photon)

> **Warning**
> Do not merge until #30050 has been merged.

## Proposed changes:

The ActivityPub plugin returns an object with information about each post when you make a ActivityPub request. This includes an image for the post, fetched using core's `wp_get_attachment_image_src`.

Jetpack's Image CDN changes the URL of the images returned by that function. In some scenarios, that may not be desirable (Some Fediverse instances may not always be able to fetch images from the Jetpack CDN, for example).

In this PR, we add a new compatibility layer for the ActivityPub plugin, where we allow disabling Jetpack's Image CDN processing when fetching an image for ActivityPub. This is only enabled if you use a filter:

```php
add_filter( 'jetpack_activitypub_post_disable_cdn', '__return_true' );
```

**Discussions**

- https://github.com/pfefferle/wordpress-activitypub/pull/307
- https://github.com/pfefferle/wordpress-activitypub/pull/309

### Other information:


## Jetpack product discussion

* Internal reference: p1682284516722569-slack-C04TJ8P900J

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Go to Jetpack > Settings > Performance and ensure that the Site Accelerator feature is enabled.
* Install and activate the current development version of the ActivityPub plugin, from https://github.com/automattic/wordpress-activitypub/
* Publish a post that includes an image.
* Make a GET request for that post, with the `Accept: application/activity+json` header.
* You should see the image using a `i0.wp.com` prefix in the response
* Now add the above snippet in a functionality plugin, or in your theme's `functions.php` file.
Make the same request again; the `i0.wp.com` prefix should be gone.
